src/ducks/apps/components/Application.jsx

```diff
@@ -0,0 +1,49 @@
+import React, { Component } from 'react'
+import { Route } from 'react-router-dom'
+
+import { translate } from 'cozy-ui/react/I18n'
+import Spinner from 'cozy-ui/react/Spinner'
+
+import ApplicationDetails from './ApplicationDetails'
+import UninstallModal from './UninstallModal'
+
+export class ApplicationComponent extends Component {
+  openApp (related) {
+    window.location.href = related
+  }
+
+  render () {
+    const { t, installedApps, isFetching, fetchError, parent } = this.props
+    return (
+      <div>
+        {
+          isFetching
```

Comment:
It seems to be a good idea to put these things apart but I think it's confusing. It's not really a component but a way to put in common some component code (fetching status, error message, routing). Maybe we could keep the loading and the error part on each components and rename this part ApplicationRouting?
Also this part will still be copy/pasted in the discover component for now, won't it?